### PR TITLE
feat!: allow JSDoc tags without explicit value (e.g. `@deprecated`) to default to `true`

### LINKED ISSUE #1012: `@deprecated` should be a boolean
Thank you for the tool, it is very useful.

I just updated to 0.97.0 today and had unexpected changes to the schema.

Related to #972

```ts
/**
* @deprecated
*/
type MyType = string;
```

Should result in:

```js
   "deprecated": true
```

Not

```js
   "deprecated": ""
```

See: [Generic keywords — Understanding JSON Schema 2020-12 documentation](https://json-schema.org/understanding-json-schema/reference/generic.html) scroll down to `New in draft 2019-09`

### PR DESCRIPTION
**Pull request by mrazauskas**
Fixes #1012

As it is mentioned in the issue, currently a tag without explicit value defaults to an empty string. For example, `@deprecated` results in: `"deprecated": ""`.

What if tags without value would default to `true` instead? On surface this feels like right pattern which reflects actual usage. For instance, this generator has own `@nullable` tag, which is assumed to be `@nullable true`. Similarly TypeScript has `@deprecated`, `@internal`.

Also I added tests for `@readOnly`, `@writeOnly`, which could work in the same way.

Just an idea. Could anything get broken?

# Version

Published prerelease version: `v0.98.1-next.4`

<details>
  <summary>Changelog</summary>

  :tada: This release contains work from new contributors! :tada:
  
  Thanks for all your work!
  
  :heart: Tom Mrazauskas ([@mrazauskas](https://github.com/mrazauskas))
  
  :heart: Jason Dent ([@Jason3S](https://github.com/Jason3S))
  
  :heart: Remi Cattiau ([@loopingz](https://github.com/loopingz))
  
  :heart: Hadrien Milano ([@hmil](https://github.com/hmil))
  
  #### 🚀 Enhancement
  
  - feat: support key remapping via `as` [#1174](https://github.com/vega/ts-json-schema-generator/pull/1174) ([@Jason3S](https://github.com/Jason3S))
  - feat: support intrinsic string manipulation types [#1173](https://github.com/vega/ts-json-schema-generator/pull/1173) ([@Jason3S](https://github.com/Jason3S))
  - feat: support template literals as types. [#1171](https://github.com/vega/ts-json-schema-generator/pull/1171) ([@Jason3S](https://github.com/Jason3S))
  - feat: move private to protected to allow extending behavior [#1161](https://github.com/vega/ts-json-schema-generator/pull/1161) ([@loopingz](https://github.com/loopingz))
  - feat: add support for never type [#1154](https://github.com/vega/ts-json-schema-generator/pull/1154) ([@hmil](https://github.com/hmil))
  
  #### 🐛 Bug Fix
  
  - feat!: allow JSDoc tags without explicit value (e.g. `@deprecated`) to default to `true` [#1172](https://github.com/vega/ts-json-schema-generator/pull/1172) ([@mrazauskas](https://github.com/mrazauskas))
  
  #### 🔩 Dependency Updates
  
  - chore(deps-dev): bump @typescript-eslint/eslint-plugin from 5.14.0 to 5.15.0 [#1164](https://github.com/vega/ts-json-schema-generator/pull/1164) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @babel/core from 7.17.5 to 7.17.8 [#1162](https://github.com/vega/ts-json-schema-generator/pull/1162) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump auto from 10.34.1 to 10.34.2 [#1163](https://github.com/vega/ts-json-schema-generator/pull/1163) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump prettier from 2.5.1 to 2.6.0 [#1165](https://github.com/vega/ts-json-schema-generator/pull/1165) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps): bump @types/json-schema from 7.0.9 to 7.0.10 [#1166](https://github.com/vega/ts-json-schema-generator/pull/1166) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @auto-it/conventional-commits from 10.34.1 to 10.34.2 [#1167](https://github.com/vega/ts-json-schema-generator/pull/1167) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @typescript-eslint/parser from 5.14.0 to 5.15.0 [#1168](https://github.com/vega/ts-json-schema-generator/pull/1168) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps): bump commander from 9.0.0 to 9.1.0 [#1169](https://github.com/vega/ts-json-schema-generator/pull/1169) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @auto-it/first-time-contributor from 10.34.1 to 10.34.2 [#1170](https://github.com/vega/ts-json-schema-generator/pull/1170) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @typescript-eslint/eslint-plugin from 5.13.0 to 5.14.0 [#1157](https://github.com/vega/ts-json-schema-generator/pull/1157) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump vega from 5.21.0 to 5.22.0 [#1156](https://github.com/vega/ts-json-schema-generator/pull/1156) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @typescript-eslint/parser from 5.13.0 to 5.14.0 [#1158](https://github.com/vega/ts-json-schema-generator/pull/1158) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump eslint from 8.10.0 to 8.11.0 [#1159](https://github.com/vega/ts-json-schema-generator/pull/1159) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @typescript-eslint/parser from 5.12.1 to 5.13.0 [#1144](https://github.com/vega/ts-json-schema-generator/pull/1144) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump auto from 10.32.6 to 10.34.1 [#1145](https://github.com/vega/ts-json-schema-generator/pull/1145) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @typescript-eslint/eslint-plugin from 5.12.1 to 5.13.0 [#1146](https://github.com/vega/ts-json-schema-generator/pull/1146) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump ts-node from 10.5.0 to 10.7.0 [#1147](https://github.com/vega/ts-json-schema-generator/pull/1147) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @auto-it/conventional-commits from 10.32.6 to 10.34.1 [#1148](https://github.com/vega/ts-json-schema-generator/pull/1148) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps): bump typescript from 4.5.5 to 4.6.2 [#1149](https://github.com/vega/ts-json-schema-generator/pull/1149) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @auto-it/first-time-contributor from 10.32.6 to 10.34.1 [#1150](https://github.com/vega/ts-json-schema-generator/pull/1150) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump eslint-config-prettier from 8.4.0 to 8.5.0 [#1151](https://github.com/vega/ts-json-schema-generator/pull/1151) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps): bump actions/checkout from 2 to 3 [#1143](https://github.com/vega/ts-json-schema-generator/pull/1143) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @typescript-eslint/eslint-plugin from 5.12.0 to 5.12.1 [#1139](https://github.com/vega/ts-json-schema-generator/pull/1139) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump eslint from 8.9.0 to 8.10.0 [#1138](https://github.com/vega/ts-json-schema-generator/pull/1138) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @types/node from 17.0.18 to 17.0.21 [#1140](https://github.com/vega/ts-json-schema-generator/pull/1140) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @types/jest from 27.4.0 to 27.4.1 [#1141](https://github.com/vega/ts-json-schema-generator/pull/1141) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @typescript-eslint/parser from 5.12.0 to 5.12.1 [#1142](https://github.com/vega/ts-json-schema-generator/pull/1142) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps): bump actions/setup-node from 2.5.1 to 3 [#1137](https://github.com/vega/ts-json-schema-generator/pull/1137) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @typescript-eslint/parser from 5.11.0 to 5.12.0 [#1132](https://github.com/vega/ts-json-schema-generator/pull/1132) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @babel/core from 7.17.2 to 7.17.5 [#1133](https://github.com/vega/ts-json-schema-generator/pull/1133) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @types/node from 17.0.17 to 17.0.18 [#1134](https://github.com/vega/ts-json-schema-generator/pull/1134) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @typescript-eslint/eslint-plugin from 5.11.0 to 5.12.0 [#1135](https://github.com/vega/ts-json-schema-generator/pull/1135) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump eslint-config-prettier from 8.3.0 to 8.4.0 [#1136](https://github.com/vega/ts-json-schema-generator/pull/1136) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  
  #### Authors: 5
  
  - [@dependabot[bot]](https://github.com/dependabot[bot])
  - Hadrien Milano ([@hmil](https://github.com/hmil))
  - Jason Dent ([@Jason3S](https://github.com/Jason3S))
  - Remi Cattiau ([@loopingz](https://github.com/loopingz))
  - Tom Mrazauskas ([@mrazauskas](https://github.com/mrazauskas))
</details>

### CODE REVIEW

File: src/AnnotationsReader/BasicAnnotationsReader.ts [hunk 1]
```diff
@@ -85,7 +85,9 @@ export class BasicAnnotationsReader implements AnnotationsReader {
     }
 
     private parseJsDocTag(jsDocTag: ts.JSDocTagInfo): any {
-        const text = (jsDocTag.text ?? []).map((part) => part.text).join("");
+        // Tags without explicit value (e.g. `@deprecated`) default to `true`.
+        const text = jsDocTag.text ? jsDocTag.text.map((part) => part.text).join("") : "true";
```
Comment (domoritz) [inline src/AnnotationsReader/BasicAnnotationsReader.ts]: Use optional chaining.

**Review by domoritz — APPROVED** [review]
Looks good but should probable be a feature so we bump the version accordingly.

**Comment by Jason3S** [comment]
@mrazauskas,

Thank you!

**Comment by domoritz** [comment]
Thank you!

**Comment by mrazauskas** [comment]
Thanks for quick review!

**Pull request merged**